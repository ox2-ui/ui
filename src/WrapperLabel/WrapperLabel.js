import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin-top: 15px;
  padding-bottom: 5px;
`;

const Label = styled.div`
  font-size: 13.5px;
  font-family: Roboto, sans-serif;
  color: hsla(0, 0%, 35%, 1);
  padding-bottom: 5px;
`;

/**
 * WrapperLabel Component
 */
const WrapperLabel = ({ children, text }) => (
  <Wrapper>
    <Label>{text}</Label>
    {children}
  </Wrapper>
);


WrapperLabel.propTypes = {
  children: PropTypes.node.isRequired,
  text: PropTypes.string.isRequired,
};

export default WrapperLabel;
