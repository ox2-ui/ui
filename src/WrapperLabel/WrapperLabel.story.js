import React from 'react';
import { storiesOf } from '@kadira/storybook';
import WrapperLabel from './WrapperLabel';

storiesOf('WrapperLabel', module)
  .add('default', () => (
    <WrapperLabel text="My label">
      <input type="text" />
    </WrapperLabel>
  ));
