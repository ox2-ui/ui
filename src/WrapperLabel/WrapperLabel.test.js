/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import WrapperLabel from './WrapperLabel';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <WrapperLabel text="My label">
      <input type="text" />
    </WrapperLabel>
  );

  expect(wrapper).toMatchSnapshot();
});
