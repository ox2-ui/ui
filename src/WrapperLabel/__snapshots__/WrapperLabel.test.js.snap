exports[`test renders correctly 1`] = `
<WrapperLabel
  text="My label">
  <styled.div>
    <div
      className="hgoCNT">
      <styled.div>
        <div
          className="eFyCaC">
          My label
        </div>
      </styled.div>
      <input
        type="text" />
    </div>
  </styled.div>
</WrapperLabel>
`;
